crates/services/gas_price_service/src/static_updater.rs

```diff
@@ -0,0 +1,40 @@
+use crate::UpdateAlgorithm;
+use async_trait::async_trait;
+use fuel_core_types::fuel_types::BlockHeight;
+
+pub struct StaticAlgorithmUpdater {
+    static_price: u64,
+}
+
+impl StaticAlgorithmUpdater {
+    pub fn new(static_price: u64) -> Self {
+        Self { static_price }
+    }
+}
+
+pub struct StaticAlgorithm {
+    price: u64,
+}
+
+impl StaticAlgorithm {
+    pub fn new(price: u64) -> Self {
+        Self { price }
+    }
+
+    pub fn price(&self) -> u64 {
+        self.price
+    }
+}
+
+#[async_trait]
+impl UpdateAlgorithm for StaticAlgorithmUpdater {
+    type Algorithm = StaticAlgorithm;
+
+    fn start(&self, _for_block: BlockHeight) -> Self::Algorithm {
+        StaticAlgorithm::new(self.static_price)
+    }
+
+    async fn next(&mut self, _for_block: BlockHeight) -> Self::Algorithm {
```

Comment:
Maybe because we don't use `_for_block` it is better to remove for now and add then we really need it